### ISSUE TITLE
invalid_regex points at the wrong place when raw string is used.

### ISSUE THREAD

**Issue by kennytm** [rust-lang-nursery/rust-clippy#2220]
Test case:

```rust
extern crate regex;
use regex::Regex;
fn main() {
    let _ = Regex::new(r"[...\/...]").unwrap();
    let _ = Regex::new(r#"[...\/...]"#).unwrap();
    let _ = Regex::new(r#####"[...\/...]"#####).unwrap();
}
```

This gives the following lints:

```
error: regex syntax error: unrecognized escape sequence
 --> src/main.rs:4:29
  |
4 |     let _ = Regex::new(r"[...\/...]").unwrap();
  |                             ^
  |
  = note: #[deny(invalid_regex)] on by default
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.169/index.html#invalid_regex

error: regex syntax error: unrecognized escape sequence
 --> src/main.rs:5:29
  |
5 |     let _ = Regex::new(r#"[...\/...]"#).unwrap();
  |                             ^
  |
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.169/index.html#invalid_regex

error: regex syntax error: unrecognized escape sequence
 --> src/main.rs:6:29
  |
6 |     let _ = Regex::new(r#####"[...\/...]"#####).unwrap();
  |                             ^
  |
  = help: for further information visit https://rust-lang-nursery.github.io/rust-clippy/v0.0.169/index.html#invalid_regex
```

Note the column number and where the `^` is pointing at. It should be pointing at the `\`. The leading `r####` is not considered when calculating the column location.

<details><summary>clippy v0.0.169, rustc nightly 2017-11-07</summary>

```
$ cargo clippy --version
0.0.169

$ rustc -vV
rustc 1.23.0-nightly (ee2286149 2017-11-07)
binary: rustc
commit-hash: ee2286149a5f0b148334841d4f067dc819dcca3b
commit-date: 2017-11-07
host: x86_64-apple-darwin
release: 1.23.0-nightly
LLVM version: 4.0
```

</details>
